### DISCUSSION TITLE
supervised regression in mlr3spatial

### DISCUSSION THREAD

**Discussion by selvaje** [mlr-org/mlr3spatial#45] [Q&A]
Hi 
Thanks for developing the  mlr3spatial. 
I have one question. 

I'm trying to apply a supervised regression following the example at 
https://mlr3spatial.mlr-org.com/articles/mlr3spatial.html
In my case the response variable is a presence-absence (0/1) raster file with many NA values.

`task = as_task_regr(backend, target = "presence-absence")`

```
print(task)
#> <TaskRegr:backend> (122848 x 6)
#> * Target: presence-absence
#> * Properties: -
#> * Features (5):
#>   - dbl (5): band2, band3, band4, band5, band6

```
I want to use the presence-absence to do a supervised regression, therefore I can note use the 

`row_ids = sample(1:task$nrow, 500)`

Rather I need to extract the row_ids where  
presence-absence != NA
in order to fill the 

`learner$train(task, row_ids = row_ids)`

I have tried to search for an example of how to return the 
row_ids within a raster condition  but I could not find one.
Can anyone help me?
Would be nice to have also this supervised example in the https://mlr3spatial.mlr-org.com/articles/mlr3spatial.html help page.

Thanks

**Comment by pat-s** [comment]
> I'm trying to apply a supervised regression following the example at
[mlr3spatial.mlr-org.com/articles/mlr3spatial.html](https://mlr3spatial.mlr-org.com/articles/mlr3spatial.html)
In my case the response variable is a presence-absence (0/1) raster file with many NA values.

If your response variable is binary, why are you creating a regression task?

> Rather I need to extract the row_ids where
presence-absence != NA
in order to fill the

It depends a bit on what you want to achieve.
In general your task can contain NAs as long as your algorithm can handle it. If not, you might need to apply some preprocessing to filter them out. 

Applying a simple train/test split is just a toy example here, I don't recommend to use it for real applications. You might rather want to use cross-validation to estimate model performance (if that's your goal).

I might be wrong but I think your problem is not really a "spatial" one or one related to {mlr3spatial} but a general "how handle NA in your dataset" question. {mlr3spatial} just helps transforming your raster dataset into a mlr3 `Task`.

Does that help?

**Comment by selvaje** [comment]
Hi Patrick thanks for you replay. 

The problem arise also if I will do a classification or a regression. 
In supervised learning the observation points are spread in spatial domain, therefore when i transform to raster  I have NA where the is no observations.

Such raster become my response variable defined as "task"
At this points I need to extract the row ID as you did in 

`row_ids = sample(1:task$nrow, 500)`
 
The only difference is that I need to extract excluding the NA and get the 0 or 1 IDs.

This is what is blocking me, and this will be the typical example of doing supervised regression or classification in spatial domain. 

The rest is clear and NA will not be present in the underline predictors (band2, band3, band4, band5, band6).
I hope that is clear now 
Thanks for your 
Help
Giuseppe

**Reply by pat-s** [reply]
*Appreciated if you could reply to my messages instead of posting an answer :)*

> In supervised learning the observation points are spread in spatial domain, therefore when i transform to raster I have NA where the is no observations.

I am still not sure about your setup and/or issues.
There are two scenarios:

- You have point data and extract information from a raster to these points. 
- You start with a raster and convert the pixels to point information

Does https://github.com/mlr-org/mlr3spatial/discussions/43 help? 

> At this points I need to extract the row ID as you did in

I am not sure why you need the row IDs and how this relates to your problem - I think I have not understood it yet. Maybe @be-marc can help?
Is your issue the presence of NAs in your response variable?
I still think you *might* want to transform your data manually into points and deal with the NAs before you start with mlr3.

Also a reprex showcasing what you want to do might shine more light on your use case. You could use a modified version of `mlr3spatial::demo_stack_spatraster()` as shown in https://github.com/mlr-org/mlr3spatial/discussions/43 to create a reprex.
